jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -783,11 +783,23 @@ ecma_op_function_try_to_lazy_instantiate_property (ecma_object_t *object_p, /**<
   if (ecma_compare_ecma_string_to_magic_id (property_name_p, LIT_MAGIC_STRING_CALLER)
       || ecma_compare_ecma_string_to_magic_id (property_name_p, LIT_MAGIC_STRING_ARGUMENTS))
   {
-    ecma_extended_object_t *ext_func_p = (ecma_extended_object_t *) object_p;
-
     const ecma_compiled_code_t *bytecode_data_p;
-    bytecode_data_p = ECMA_GET_INTERNAL_VALUE_POINTER (const ecma_compiled_code_t,
-                                                       ext_func_p->u.function.bytecode_cp);
+#ifndef CONFIG_DISABLE_ES2015_ARROW_FUNCTION
+    if (ecma_get_object_type (object_p) == ECMA_OBJECT_TYPE_ARROW_FUNCTION)
+    {
+      ecma_arrow_function_t *arrow_func_p = (ecma_arrow_function_t *) object_p;
+      bytecode_data_p = ECMA_GET_NON_NULL_POINTER (const ecma_compiled_code_t,
+                                                   arrow_func_p->bytecode_cp);
+    }
+    else
+    {
+#endif /* CONFIG_DISABLE_ES2015_ARROW_FUNCTION */
```

Comment:
I don't really like when `{}` blocks are enclosed in ifs. It would be good to avoid it somehow. Even the original code could be duplicated here.

Reply:
Quite honestly I don't know which one of those two is the lesser evil, I'd stick with the current solution, as it contains less code duplication.

Reply:
IMHO, it is better in the current form.